pkg/agent/controller/egress/egress_controller.go

```diff
@@ -466,6 +538,32 @@ func (c *EgressController) syncEgress(egressName string) error {
 		return err
 	}
 
+	localNodeSelected, err := c.cluster.ShouldSelectEgress(egress)
```

Comment:
I think you need to consider egressIP == "" (not allocated yet):
```
E0701 10:04:10.461336       1 egress_controller.go:320] Error syncing Egress egress, requeuing. Error: failed to add ip <nil> to interface eno3: numerical result out of range
E0701 10:04:11.558887       1 egress_controller.go:320] Error syncing Egress egress, requeuing. Error: failed to add ip <nil> to interface eno3: numerical result out of range
E0701 10:06:51.560270       1 egress_controller.go:320] Error syncing Egress egress, requeuing. Error: failed to add ip <nil> to interface eno3: numerical result out of range
E0701 10:11:51.560844       1 egress_controller.go:320] Error syncing Egress egress, requeuing. Error: failed to add ip <nil> to interface eno3: numerical result out of range
```

Reply:
Should the ipallocator allocate an IP for Egress with a specific ExternalIpPool?

Reply:
This is asynchronous process, and an IP pool can be full. It's totally legal to see an Egress that doesn't have an IP.

Reply:
done